Alicerce/Sources/DeepLinking/Router.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  Router.swift
+//  Alicerce
+//
+//  Created by André Pacheco Neves on 10/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import UIKit
+
+protocol RouteHandler {
+    func handle(parameters: [String : String])
+}
+
+protocol Router {
+    associatedtype Handler: RouteHandler
+
+    func register(_ route: URL, handler: Handler) throws
+    func unregister(_ route: URL) throws -> Handler
+
+    func route(_ route: URL) throws
+}
+
+final class TreeRouter<Handler: RouteHandler>: Router {
+
+    typealias Match = Route.Tree<Handler>.Match
+    private typealias ParsedRoute = (scheme: Route.Scheme, components: [Route.Component])
+
+    enum Error: Swift.Error {
+        case invalidRoute(Swift.Error)
+        case routeNotFound
+        //case routeAlreadyHandled(route: [Component], handler: Handler)
```

Comment:
I forgot to remove this 😅 

It has been fixed, since more errors have been added 😉